### PR TITLE
Gateway unsubscribe for multiple subscribed clients

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

Since we  (can) have multiple accounts/addresses registered per user and Obscuro Gateway subscribes for all of them. We return  new subscriptionID. And in case users wants to unsubscribe we want to make sure the gateway also unsubscribes from all subscriptions to the node that are/were related to that specific subscriptionID.

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: tools/walletextension/accountmanager/account_manager.go [hunk 1]
```diff
@@ -73,6 +75,17 @@ func (m *AccountManager) ProxyRequest(rpcReq *wecommon.RPCRequest, rpcResp *inte
 		}
 		return nil
 	}
+	if rpcReq.Method == rpc.Unsubscribe {
+		if len(rpcReq.Params) != 1 {
+			return fmt.Errorf("one parameter (subscriptionID) expected, %d parameters received", len(rpcReq.Params))
+		}
+		subscriptionID, ok := rpcReq.Params[0].(string)
+		if !ok {
+			return fmt.Errorf("subscriptionID needs to be a string. Got: %d", rpcReq.Params[0])
```
Comment (otherview) [inline tools/walletextension/accountmanager/account_manager.go:84]:
```suggestion
			return fmt.Errorf("subscriptionID needs to be a string. Got: %v", rpcReq.Params[0])
```

File: tools/walletextension/subscriptions/subscriptions.go [hunk 2]
```diff
@@ -159,3 +183,20 @@ func prepareLogResponse(idAndLog common.IDAndLog, userSubscriptionID gethrpc.ID)
 	}
 	return jsonResponse, nil
 }
+
+func (sm *SubscriptionManager) HandleUnsubscribe(userSubscriptionID string, rpcResp *interface{}) {
+	subscriptions, exists := sm.subscriptionMappings[userSubscriptionID]
+
+	if !exists {
```
Comment (otherview) [inline tools/walletextension/subscriptions/subscriptions.go:190]:
```suggestion
	subscriptions, exists := sm.subscriptionMappings[userSubscriptionID]
	if !exists {
```

File: tools/walletextension/subscriptions/subscriptions.go [hunk 3]
```diff
@@ -106,39 +100,69 @@ func readFromChannelAndWriteToUserConn(channel chan common.IDAndLog, userConn us
...
+func (sm *SubscriptionManager) unsubscribeAndRemove(userSubscriptionID string, subscription *gethrpc.ClientSubscription) {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	subscription.Unsubscribe()
+
+	subscriptions, exists := sm.subscriptionMappings[userSubscriptionID]
+	if !exists {
+		return
+	}
```
Comment (otherview) [inline tools/walletextension/subscriptions/subscriptions.go:112]: Is this possible to happen ? that a subscription is cancelled but there isn't one in our mapping ?

**Review by otherview — APPROVED** [review]
Minor bits nothing blocking imo

File: tools/walletextension/subscriptions/subscriptions.go (hunk 3 above)
Comment (zkokelj) [reply tools/walletextension/subscriptions/subscriptions.go:112]: Should not happen, added it here to be 100% sure not to crash. I added a log that will clearly tell us if & what happened.

**Pull request merged**